pubsub/spec/sample_spec.rb

```diff
@@ -0,0 +1,166 @@
+# Copyright 2015 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative "../sample"
+require "spec_helper"
+
+describe "Pub/Sub sample" do
+  before(:context) do
+    @gcloud = Gcloud.new ENV["GCLOUD_PROJECT"]
+    @pubsub = @gcloud.pubsub
+    @topic_name = "my-topic"
+  end
+
+  before(:each) do
+    $stdout = StringIO.new
+  end
+
+  it "creates topic" do
+    topic = @pubsub.topic @topic_name
+    topic.delete unless topic.nil?
+
+    allow(Gcloud).to receive(:new).with("my-gcp-project-id").and_return(@gcloud)
+
+    create_topic
+
+    expect($stdout.string).to include(@topic_name)
```

Comment:
```ruby
expect { create_topic }.to output("Topic created my-topic").to_stdout
```

(Using built-in [`output` matcher](https://www.relishapp.com/rspec/rspec-expectations/v/3-4/docs/built-in-matchers/output-matcher))